WD-Firmware/examples/ex-freertos/main.c

```diff
@@ -225,19 +227,24 @@ void demo_init(void *pMem)
 	TimerHandle_t xExampleSoftwareTimer = NULL;
 #else
 	u32_t res;
+	pspExceptionCause_t cause;
 #endif /* D_USE_RTOSAL */
 
     /* Configure the system ready to run the demo.  The clock configuration
     can be done here if it was not done before main() was called. */
     prvSetupHardware();
 
 #ifdef D_USE_RTOSAL
-    /* install exception handler */
-    rtosalInstallExceptionIsr(ulSynchTrap);
-    /* install timer interrupt handler */
-    rtosalInstallIsr(vPortSysTickHandler, E_MACHINE_TIMER_SOURCE_INT);
+    /* register exception handlers */
+    for (cause = E_EXC_INSTRUCTION_ADDRESS_MISALIGNED ; cause < E_EXC_LAST ; cause++)
+    {
+        pspRegisterIsrExceptionHandler(vSynchTrapUnhandled, cause);
+    }
+    pspRegisterIsrExceptionHandler(vSynchTrap, E_EXC_ENVIRONMENT_CALL_FROM_MMODE);

```

Comment:
can you please add a comment here explaing which exception you are registering ?
pspRegisterIsrExceptionHandler(vSynchTrap, E_EXC_ENVIRONMENT_CALL_FROM_MMODE);

Reply:
Done